### ISSUE TITLE
Preview with serve looks perfect; format is weird on live website (may be dependency-related, may not be)

### ISSUE THREAD

**Issue by griffinjmbur** [academicpages/academicpages.github.io#1919]
**Describe the bug**
Hi, folks. I'm more-or-less a n00b and have limited HTML experience so forgive me if I am missing something (mostly do datascience/sociology stuff). I have managed to get a very nice looking local preview, but when I push to GH, it looks funky and almost entirely unformatted. The two attached images show the difference. 
<img width="945" alt="Screenshot 2024-02-16 at 4 30 44 PM" src="https://github.com/academicpages/academicpages.github.io/assets/126503459/cd0b7900-d020-43d1-8513-e7804642ede8">
<img width="861" alt="Screenshot 2024-02-16 at 4 29 00 PM" src="https://github.com/academicpages/academicpages.github.io/assets/126503459/d1ae495d-c38d-47f1-ad81-67b2c7eecc2c">

I have mostly followed the steps here (https://archive.ph/Tnzxg), but one possible source of error is that I write in Mathjax in a Python Jupyter notebook, so I did need to manually add a mathjax-supplement file in _includes and then modify the default.html file in _layouts, which was successful locally. I don't know if that could possibly cause this issue, though.

The error I see (technically an annotation/warning) in the Actions log is this: 

> build
> The github-pages gem can't satisfy your Gemfile's dependencies. If you want to use a different Jekyll version or need additional dependencies, consider building Jekyll site with GitHub Actions: https://jekyllrb.com/docs/continuous-integration/github-actions/

I did try using a GitHub Actions workflow, but to no avail (I just used their suggested default .yml file for Jekyll). 

I also saw some recent issues involving the "excerpt" tags that caused people trouble, but I only cloned this repo very recently, and, perhaps for that reason, I actually don't see any "excerpt" tags in the pre-shipped markdown files to begin with. 

Lastly, I also get this warning in the terminal when I push:

> GitHub Metadata: No GitHub API authentication could be found. Some fields may be missing or have incorrect data.

I haven't tried to mess with that since after Googling it appeared to be mostly an informational message per other users. I also was able to successfully publish a test run of this site using minimal-mistakes earlier today, and this wasn't a problem. 

I greatly appreciate any help or thoughts that anyone has in advance.

**Comment by rjzupkoii** [comment]
This appears to be the same issue as #1041 - can you link to one of the builds that got pushed to your live website?

> GitHub Metadata: No GitHub API authentication could be found. Some fields may be missing or have incorrect data.

This warning you can ignore in the terminal. At some point I'll take a closer look at it to see if there's a way to suppress it since it doesn't impact local testing.

**Comment by griffinjmbur** [comment]
Forgive me if I'm not linking the correct thing, but I think this should be a good example of a build that worked locally to include the theme but does not include it live. 

https://github.com/griffinjmbur/griffinjmbur.github.io/actions/runs/7939272300

By the way, I should note that, although I cannot remember where, I did remember between my first post and this one that I *have* read about the mathjax fix specifically causing themes not to display. I'm not sure if that's helpful. 

I can directly post the specifics of my Mathjax modifications if useful, but they basically followed the guidelines here (https://archive.ph/aKZ29).

**Comment by rjzupkoii** [comment]
@griffinjmbur Thanks for those links! I still haven't been able to reproduce the error on my slide; however, I suspect there might be a conflict between how MathJax works on stock Minimal Mistakes and Academic Pages. I've bumped MathJax to version 3.0 and updated the [template's guide page](https://academicpages.github.io/markdown/) to demonstrate things up and running. You'll likely need to undo the MathJax related changes in griffinjmbur/griffinjmbur.github.io@47dea63 but hopefully this fixes the problem!

**Comment by rjzupkoii** [comment]
@griffinjmbur From what I was reading it seems like some approaches to MathJax end up overriding some of the template's settings, so that's most likely what is happening. With MathJax version 3.0 you really shouldn't need much more that the two lines in green from 45ea6b2 - I'd recommend starting by undoing anything you did related to MathJax to see if you can start by just getting the site deployed successfully. Then once you have that you can try adding these two lines to `_includes/head/custom.html`:

```html
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script id="MathJax-script" async src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
```

**Comment by griffinjmbur** [comment]
Ah, I see. Thanks for the tip. 

I removed the Mathjax supplemental file and the reference to it in the default. Still the same problem, although I may have forgotten somewhere else that I modified the files to get Mathjax working. 

I'm going to try to wipe it all clean and start over and see if I can just use your slight modifications to the code mentioned above locally and then try pushing. At this point, I can't seem to get the theme to look coherent on the website even without adding any posts of my own or any reference to MJ. Thanks for the help! I'll report back.
